hazelcast/src/main/java/com/hazelcast/spi/impl/servicemanager/RemoteServiceDescriptor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.servicemanager;
+
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.RemoteService;
+
+/***
+ * Descriptor of Hazelcast remote service
```

Comment:
Can you add some documentation how this class fits into the big picture? 

Also this interface AFAIK is public API, because if a user wants to integrate his service, like TreeService, he needs to come up with an implementation.

And do we need an interface at all here? I think that RemoteServiceDescriptor can better be modified as a final class since it has just 2 fields. The only reason you need this class is because you need to have a pair <ServiceName,RemoteService>. So it is just a 'dumb' record.